src/main/java/io/confluent/connect/hdfs/hive/HiveMetaStore.java

```diff
@@ -0,0 +1,344 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.connect.hdfs.hive;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.AlreadyExistsException;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.InvalidObjectException;
+import org.apache.hadoop.hive.metastore.api.InvalidOperationException;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.UnknownDBException;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.apache.hive.hcatalog.common.HCatUtil;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.confluent.connect.hdfs.HdfsSinkConnectorConfig;
+import io.confluent.connect.hdfs.errors.HiveMetaStoreException;
+
+public class HiveMetaStore {
+
+  private static final Logger log = LoggerFactory.getLogger(HiveMetaStore.class);
+  private final IMetaStoreClient client;
+
+  public HiveMetaStore(Configuration conf, HdfsSinkConnectorConfig connectorConfig) throws HiveMetaStoreException {
+    HiveConf hiveConf = new HiveConf(conf, HiveConf.class);
+    String hiveConfDir = connectorConfig.getString(HdfsSinkConnectorConfig.HIVE_CONF_DIR_CONFIG);
+    hiveConf.addResource(new Path(hiveConfDir, "hive-site.xml"));
+    try {
+      client = HCatUtil.getHiveMetastoreClient(hiveConf);
+    } catch (IOException | MetaException e) {
+      throw new HiveMetaStoreException(e);
+    }
+  }
+
+  private interface ClientAction<R> {
+    R call() throws TException;
+  }
+
+  private <R> R doWithRetry(ClientAction<R> action) throws TException {
+    try {
+      return action.call();
+    } catch (TException e) {
```

Comment:
Is any logging here worthwhile? It'd probably be good to give some indication if the connection is continuously failing.